Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityHelpers.cpp

```diff
@@ -213,6 +213,18 @@ namespace AzToolsFramework
         return componentClassData;
     }
 
+    AZStd::string GetNameFromComponentClassData(const AZ::Component* component)
+    {
+        const char* className = component->RTTI_GetTypeName();
+        const AZ::SerializeContext::ClassData* classData = GetComponentClassData(component);
+        if (!classData)
+        {
+            return className;
+        }
```

Comment:
This is a safety check. If you look at GetComponentClassData(), you'll see that it's fetching the class data for the underlying component type and it can be a nullptr. So, in the off chance that it happens, we are using fallback logic to give it the typename of component. 

An example where this can can happen if a component forgets to reflect its data to serializecontext or don't want to reflect at all. In that case, this'll fetch 'GenericComponentWrapper' as the typename. Will add a comment here.